### PR TITLE
Remove Halide functionality used for generating input to PPCG

### PR DESCRIPTION
**Pull request by skimo-openhub**
PPCG was removed before TC went public.  This functionality is not needed anymore.

### CODE REVIEW

**Comment by skimo-openhub** [comment]
Cherry-picked from #39.

**Comment by prigoyal** [comment]
very cool, thanks a bunch @skimo-openhub this allows keeping our code and history clean.

looks good to go to me. Did you run all the gpu tests as well?

File: src/core/halide_utils.cc [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+#include "tc/core/utils/dlpack.h"
+
+namespace tc {
+
+using namespace dlutils;
+using namespace Halide;
+using namespace Halide::Internal;
+
+DLDataType fromHalideType(const Halide::Type& type) {
+  // Hmm, these are suspiciously similar...
```
Comment (abadams) [inline src/core/halide_utils.cc]: haha, tell me about it. You'll want to check that the code enum matches between Halide and DLType though.

**Comment by skimo-openhub** [comment]
On Fri, Feb 23, 2018 at 05:32:14PM +0000, Andrew Adams wrote:
> +DLDataType fromHalideType(const Halide::Type& type) {
> +  // Hmm, these are suspiciously similar...
> 
> haha, tell me about it. You'll want to check that the code enum matches between Halide and DLType though.

It looks like Zach wrote that.  Alex just renamed the file and
I just cherry-picked the commits.

skimo

**Comment by prigoyal** [comment]
merging it now, CI is green :)

**Pull request merged**